source/config/Config.cpp

```diff
@@ -1077,12 +1135,225 @@ bool PlainConfig::PubSub::Validate() const
     {
         if (!FileUtils::ValidateFilePermissions(subscribeFile.value(), Permissions::PUB_SUB_FILES, true))
         {
+            LOGM_ERROR(
+                Config::TAG,
+                "*** %s: subscribeFile field is not valid for the Pub-Sub sample feature ***",
+                DeviceClient::DC_FATAL_ERROR);
             return false;
         }
     }
     return true;
 }
 
+void PlainConfig::PubSub::SerializeToObject(Crt::JsonObject &object) const
+{
+    (void)object;
+    object.WithBool(JSON_ENABLE_PUB_SUB, enabled);
+
+    if (publishTopic)
```

Comment:
The type of this ```publishFile``` is ```Aws::Crt::Optional<std::string>``` so I added has value check ```publishFile.has_value()``` in case the filed is missing in json:

```"samples": {
		"pub-sub": {
			"enabled": true,
			"publish-topic": "publish_topic",
			"subscribe-topic": "subscribe_topic"
		}